src/containerapp/azext_containerapp/custom.py

```diff
@@ -690,6 +705,24 @@ def create_containerapp(cmd,
         else:
             set_managed_identity(cmd, resource_group_name, containerapp_def, user_assigned=[registry_identity])
     try:
+        image = None if imageNotProvided else _reformat_image(source,repo,image)
+        resource_group = ResourceGroup(cmd, name=resource_group_name, location=location)
+        env = ContainerAppEnvironment(cmd, managed_env, resource_group, location=location)
+        app = ContainerApp(cmd, name, resource_group, None, image, env, target_port, registry_server, registry_user, registry_pass, env_vars, workload_profile_name, ingress)
+
+        if source or repo:
+            _get_registry_details(cmd,app,source) # fetch ACR creds from arguments registry arguments
+
+        if source and not _has_dockerfile(source, dockerfile):
+            pass
+        else:
+          dockerfile_content = _get_dockerfile_content(repo, branch, token, source, context_path, dockerfile)
+          ingress, target_port = _get_ingress_and_target_port(ingress, target_port, dockerfile_content)
+
+        app.create_acr_if_needed()
```

Comment:
Are we able to remove this? Even if it doesn't create the ACR instance in _any_ of the flows provided by this function, if the implementation changes under-the-hood, it could start causing some unexpected behavior when we don't want the ACR instance to be created in any circumstance.

Reply:
Removed this line. Since it is now required to set registry-server, registry-user and registry-pass args, the method `create_acr_if_needed()` is not longer required